src/seastar_lw_shared_ptr.rs

```diff
@@ -0,0 +1,261 @@
+use crate::fmt::display;
+use crate::kind::Trivial;
+use crate::string::CxxString;
+use crate::ExternType;
+use core::ffi::c_void;
+use core::fmt::{self, Debug, Display};
+use core::marker::PhantomData;
+use core::mem::MaybeUninit;
+use core::ops::Deref;
+
+/// Binding to C++ `seastar::lw_shared_ptr<T>`.
+#[repr(C)]
+pub struct SeastarLwSharedPtr<T>
+where
+    T: SeastarLwSharedPtrTarget,
+{
+    repr: [MaybeUninit<*mut c_void>; 2],
```

Comment:
I assume that this array is here so that the struct has correct size. Why does it have 2 elements then? You put the following assert in `gen/src/write.rs`:
```
"static_assert(sizeof(::seastar::lw_shared_ptr<{}>) == sizeof(void *), \"\");",
```
Version for `std::shared_ptr` has `2 * sizeof(void *)` - so I assume that this array should also have just 1 element?